test/test_mps.py

```diff
@@ -1203,6 +1203,29 @@ def __exit__(self, exec_type, exec_value, traceback):
 
             raise RuntimeError(msg)
 
+class TestAutocastMPS(TestCase):
+
+    def test_matmul_autocast(self):
+        autocast_tensor_A = torch.rand((8, 8), device="mps")
+        autocast_tensor_B = torch.rand((8, 8), device="mps")
+        tensor_A = autocast_tensor_A.clone().detach()
+        tensor_B = autocast_tensor_B.clone().detach()
+        autocast_output_tensor = torch.empty(8, 8)
+        output_tensor = autocast_output_tensor.clone().detach()
+
+        with torch.autocast(device_type="mps"):
+            autocast_output_tensor = torch.mm(autocast_tensor_A, autocast_tensor_B)
+            autocast_output_tensor = torch.mm(autocast_tensor_A, autocast_output_tensor)
+
+        output_tensor = torch.mm(tensor_A, tensor_B)
+        output_tensor = torch.mm(tensor_A, output_tensor)
+
+        self.assertEqual(autocast_output_tensor.dtype, torch.float16, "Autocast output tensor was not expected type float16")
+        self.assertEqual(autocast_output_tensor,
+                         output_tensor.to(torch.float16),
```

Comment:
Is this expected to be `bfloat16` after https://github.com/pytorch/pytorch/pull/99272/commits/f282bf47427ba9b876b2ae496716921482d0e870?

Reply:
Hmm, but than autocast is only available on MacOS14 or newer, is that expected?

Reply:
Yes. I don't know if this is intentional or not.

cc @skotapati @kulinseth

Reply:
Yeah sorry, missed updating the tests after switching to bf16

Reply:
Question trying to understand whats exactly going on with mps and bf16, saw on sonoma newer m2/m3 support bf16, but with flux and some other models for training was told bf16 isnt supported on macbooks, is it just that we need newer editions of pytorch and the devs from those trainers were mistaken that bf16 is supported?

Reply:
m2+ hardware is needed and pytorch before 2.4 didn't have bf16 and pytorch after 2.4 has correctness issues on MPS. it's a sore situation and i think the safest answer for now is "don't use it"